app/javascript/grid.js

```diff
@@ -1,3 +1,44 @@
+var selected = [];
+function nfinder (x,y,c){
+var id = "block_"+x+"x"+y;
+//console.log(document.getElementById("block_"+x+"x"+y).outerHTML);
+var myElem = document.getElementById(id);
+if ((myElem !== null) && (myElem.style.backgroundColor == c) && ( selected.indexOf(id) == -1 )) {
+
+console.log("Working on block_"+x+"x"+y);
+selected.push(id); 
+
+//nfinder (x-1,y+1,c) 
+nfinder (x,y+1,c) //Top
+//nfinder (x+1,y+1,c)
+
+nfinder (x-1,y,c) //Left
+nfinder (x+1,y,c) //Right
+
+//nfinder (x-1,y-1,c)
+nfinder (x,y-1,c) //Bottom
+//nfinder (x+1,y-1,c)
+}
+}
```

Comment:
It's unreadable

Reply:
Cleaned up, i hope it will be more readable now